src/awkward/_v2/types/recordtype.py

```diff
@@ -148,6 +148,14 @@ def _str(self, indent, compact):
         return [self._str_categorical_begin()] + out + [self._str_categorical_end()]
 
     def __repr__(self):
+        if (
+            self._typestr != "override"
+            and self._parameters is not None
+            and len(self._parameters) == 1
+            and "__record__" in self._parameters
+        ):
+            args = dict(zip(self._fields, [x.primitive for x in self._contents]))
+            return "{}[{}]".format(self._parameters["__record__"], str(args)[1:-1])
         args = [repr(self._contents), repr(self._fields)] + self._repr_args()
         return "{}({})".format(type(self).__name__, ", ".join(args))
```

Comment:
This is a style thing, but I think it's better to be explicit about the if-else structure. Early returns are better when the rest of the code in the function is much longer than the early return.

```suggestion
            args = dict(zip(self._fields, [x.primitive for x in self._contents]))
            return "{}[{}]".format(self._parameters["__record__"], str(args)[1:-1])
        else:
            args = [repr(self._contents), repr(self._fields)] + self._repr_args()
            return "{}({})".format(type(self).__name__, ", ".join(args))
```